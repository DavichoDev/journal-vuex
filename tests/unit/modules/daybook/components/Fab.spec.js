import { shallowMount } from "@vue/test-utils";
import Fab from '@/modules/daybook/components/Fab.vue';

describe('Fab.vue Tests', () => {

    let wrapper
    beforeEach(() => {
      wrapper = shallowMount( Fab )
      jest.clearAllMocks()
    })
    
    test('Debe de pasar el snapshot', () => {
        expect( wrapper.html ).toMatchSnapshot()
    })
    test('Debe de mostrar el icono por defecto', () => {
        const fabIcon = wrapper.find('i')
        expect(fabIcon.classes()).toEqual(['fa', 'fa-2x'])
    })
    test('Debe de mostrar el icono por argumento: fa-plus', () => {
        wrapper = shallowMount( Fab, {
            props: {
                icon: 'fa-plus'
            }
        })
        const fabIcon = wrapper.find('i')
        expect(fabIcon.classes()).toEqual(['fa', 'fa-2x', 'fa-plus'])
    })

    test('Debe de emitir el event on:click cuando se hace click', () => {
        const fabButton = wrapper.find('button')
        fabButton.trigger('click')
        expect( wrapper.emitted('on:click') ).toHaveLength(1)
    })
})