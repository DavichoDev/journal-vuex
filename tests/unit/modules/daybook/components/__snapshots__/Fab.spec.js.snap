// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fab.vue Tests Debe de pasar el snapshot 1`] = `[Function]`;
